### PR TITLE
Add `micro-eth-signer/kzg` 

### PR DESCRIPTION
**Pull request by acolytec3**
Takes an initial pass at integrating the kzg module from `micro-eth-signer`, putting it in `tx` for now.  

I've added one of the trusted setups for now so we don't have to build/import `trusted-setups`

Also removes `loadTrustedSetup` from the kzg interface.  We don't use it anywhere and I'm not convinced we will ever need to load another one.

### CODE REVIEW

**Comment by acolytec3** [comment]
image

Here are some initial benchmarks for the various kzg functions we use.  Refer to `[root]/packages/tx/test/kzg.bench.ts` to review.

You can run with `npx vitest bench test/kzg.bench.ts` from the `tx` directory

File: packages/tx/src/jsKZGWrapper.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+//@ts-ignore  NOTE: Have placed the "fast" version of the trusted setup here for now till the `trusted-setups` package is published
+import { trustedSetup as fast } from './fast.js'
+
+import type { PrefixedHexString } from '@ethereumjs/util'
+// import { trustedSetup as fast } from 'trusted-setups/fast.js'
+
+const kzg = new KZG(fast)
+
+export const jsKZG: Kzg = {
```
Comment (holgerd77) [inline packages/tx/src/jsKZGWrapper.ts:12]: This should implement our interface from Util

**Comment by paulmillr** [comment]
The package is now live on NPM at `@paulmillr/trusted-setups`

**Comment by holgerd77** [comment]
https://github.com/paulmillr/trusted-setups

File: packages/tx/package.json [hunk 2]
```diff
@@ -57,12 +57,14 @@
     "@ethereumjs/common": "^4.4.0",
     "@ethereumjs/rlp": "^5.0.2",
     "@ethereumjs/util": "^9.1.0",
-    "ethereum-cryptography": "^3.0.0"
+    "ethereum-cryptography": "^3.0.0",
+    "trusted-setups": "github:paulmillr/trusted-setups"
```
Comment (holgerd77) [inline packages/tx/package.json:61]: This seems to have been forgotten. Will push a small update.

**Comment by holgerd77** [comment]
Have updated the `package-lock.json` file (there was a conflict) and removed the forgotten depedency.

There are some client tests failing now I also planned to fix, but I am not fully seeing through, so there is some naming/interface thing happening in the tests (e.g. `getRawBlock.spec.ts`, also others) and I am not fully sure how to fix.

File: packages/client/bin/cli.ts [hunk 3]
```diff
@@ -28,6 +28,8 @@ import {
   setLengthLeft,
   short,
 } from '@ethereumjs/util'
+//@ts-ignore trusted-setups doesn't include types
+import { trustedSetup as fast } from '@paulmillr/trusted-setups/fast.js'
```
Comment (jochem-brouwer) [inline packages/client/bin/cli.ts:32]: Nit: in tests it does not matter, but in production code I would suggest using a bit more descriptive name than `fast`. Something like `kzgSetupFast` or something.

File: packages/evm/src/opcodes/functions.ts [hunk 4]
```diff
@@ -762,7 +762,7 @@ export const handlers: Map<number, OpHandler> = new Map([
     function (runState) {
       const index = runState.stack.pop()
       if (runState.env.blobVersionedHashes.length > Number(index)) {
-        runState.stack.push(bytesToBigInt(runState.env.blobVersionedHashes[Number(index)]))
+        runState.stack.push(BigInt(runState.env.blobVersionedHashes[Number(index)]))
```
Comment (jochem-brouwer) [inline packages/evm/src/opcodes/functions.ts:765]: I will address this at some point, bit inside the EVM we should likely not put the strings in, but directly the bigints so we can push those to stack.

Comment (jochem-brouwer) [reply packages/evm/src/opcodes/functions.ts:765]: Oh, hmm ok originally they were bytes which we kept converting to bigints, which is/was also suboptimal.

File: packages/evm/vite.config.bundler.ts [hunk 5]
```diff
@@ -11,7 +11,7 @@ export default defineConfig({
       treeshake: 'safest',
     },
     lib: {
-      entry: './src/',
+      entry: '../tx/examples/blobTx.ts',
```
Comment (jochem-brouwer) [inline packages/evm/vite.config.bundler.ts:14]: This change should likely be reverted?

File: packages/util/test/kzg.bench.ts
Comment (jochem-brouwer) [inline packages/util/test/kzg.bench.ts:1]: Nice :smile:

File: packages/client/bin/cli.ts (hunk 3 above)
Comment (jochem-brouwer) [reply packages/client/bin/cli.ts:32]: Ah, just using `trustedSetup` here should be fine :) 
```suggestion
import { trustedSetup } from '@paulmillr/trusted-setups/fast.js'
```

**Review by jochem-brouwer — COMMENTED** [review]
Some super small nits, LGTM otherwise

**Review by jochem-brouwer — APPROVED** [review]
LGTM

File: packages/util/test/kzg.bench.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  })
+  describe('verifying proof', async () => {
+    bench('wasm verifyProof', () => {
+      kzg.verifyBlobKZGProofBatch([blob], [commit], [proof])
+    })
+    bench('js verifyProof', () => {
+      jsKZG.verifyBlobProofBatch([blob], [commit], [proof])
+    })
+  })
+})
```
Comment (holgerd77) [inline packages/util/test/kzg.bench.ts:36]: Ah, this is also ok if we directly put this between the tests! 🙂👍

Comment (holgerd77) [reply packages/util/test/kzg.bench.ts:36]: (maybe even better, even more „at hand“)

**Comment by holgerd77** [comment]
🎉

**Pull request merged**